### ISSUE TITLE
[feat] `react-dom/no-string-style-prop` - `allow` option

### ISSUE THREAD

**Issue by karlhorky** [Rel1cx/eslint-react#1217]
### Describe the problem

`react-dom/no-string-style-prop` looks great thanks!

However, this rule sometimes mis-reports for `style` props which are indeed strings.

For these cases it can be useful to disable this rule for some component names.

For example, Expo's `StatusBar` component uses a string `style` prop (`<StatusBar style="auto" />`) 🤔, which is reported as a problem

### Describe the solution you'd like

Would you be open to supporting [the `allow` option](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/style-prop-object.md#allow) for components to ignore?

I currently use [the `allow` option from `react/style-prop-object`](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/style-prop-object.md#allow) in my ESLint config to ignore components named `StatusBar`:

```js
  'react/style-prop-object': [
    'warn',
    {
      allow: [
        // Allow expo-status-bar style prop, which is a string,
        // eg: <StatusBar style="auto" />
        // https://github.com/expo/expo/blob/999572cd1036529ffa3a28a0490dd7c0f6f0d731/packages/expo-status-bar/src/StatusBar.types.ts#L2
        'StatusBar',
      ],
    },
  ],
```

### Alternatives considered

Alternative approaches / heuristics for auto-allowing such as a whitelist, allowing for all components, etc, each with their own downsides

### Additional context

--

**Comment by Rel1cx** [comment]
Thanks for your feedback and suggestions for improving this rule!

I find the `allow` option to be inefficient. I'd prefer this rule to only check the `style` prop on host elements. This is because, without type information, the rule can't accurately determine the type and semantics of a `style` prop on user-defined components.

While accessing type information would solve this, it would also narrow the rule's applicability, making it less useful in non-TypeScript environments like MDX.

**Comment by karlhorky** [comment]
@Rel1cx Ok, good points, thanks.

> I'd prefer this rule to only check the style prop on host elements

Oh right, that was one of my "Alternatives considered" above:

> Alternative approaches / heuristics for auto-allowing such as ... allowing for all components,

Upon consideration of your points, it appears to me to be the more correct option.

I see that https://github.com/Rel1cx/eslint-react/commit/5abf84217e3e8821ea4c9bfdb0b60787624875e0 switches the rule to only check host elements.

I'll check out the new release in our projects.

**Comment by karlhorky** [comment]
Confirmed, the checking of the `style` prop only on host elements is working 👍 Now we can migrate off `eslint-plugin-react` 🎉

- https://github.com/upleveled/eslint-config-upleveled/pull/555

Thanks!
